php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -2161,6 +2181,55 @@ abstract class NDB_BVL_Instrument extends NDB_Page
         return $dob;
     }
 
+    /**
+     * Get the candidates DOD.
+     *
+     * @return ?string The candidate's DOD.
+     */
+    function getDoD(): ?string
+    {
+        $db        = \Database::singleton();
+        $CommentID = $this->getCommentID();
+        $dod       = $db->pselectOne(
+            "SELECT c.DoD FROM flag f
+            JOIN session s ON (f.SessionID=s.ID)
+            JOIN candidate c USING (CandID)
+            WHERE f.CommentID=:CID",
+            array('CID' => $CommentID)
+        );
+        return $dod ?? null;
+    }
+
+    /**
+     * Calculates the time since candidate's death.
+     * If $date is passed, this function  will calculate the age as of
+     * the value of that argument. Otherwise the date will be calculated
+     * based on the value of the Date_taken field in the database.
+     *
+     * @param string $date (optional) The date that the age is to
+     *                     be calculated based on.
+     *
+     * @return array The time since candidate's death in an array containing
+     *         years, months and days.
```

Comment:
There's a bit of complexity here with line 2223 and it depends what you want.

That Utility function assumes that a month always has 30 days. If you're looking for a precise "time since death" then this function will give you the wrong answer.

Reply:
This is a valid point, though is the candidate age / time since death every really accurate in the case that the DOB used has format 'Ym' ? (This is the case for CCNA)

Should we be more accurate in this case?

Reply:
I'm not sure. I don't know enough about the science part of it. I think it depends what the Date of Death is being used for. I know calculateAge isn't supposed to be touched because it may impact reproducibility if we start calculating the age differently.

Maybe you could bring it up in the LORIS meeting just to be safe?

Reply:
After our discussion, I looked into the CCNA post-mortem instruments / use case and realized that this feature is never used... I may get rid of this, unless there are strong feelings about keeping it ?

Reply:
Sure, if there's no use for it.